source/extensions/filters/network/redis_proxy/command_splitter_impl.h

```diff
@@ -73,15 +74,21 @@ class CommandHandlerBase {
 
 class SplitRequestBase : public SplitRequest {
 protected:
-  static void onWrongNumberOfArguments(SplitCallbacks& callbacks, const Common::Redis::RespValue& request);
+  static void onWrongNumberOfArguments(SplitCallbacks& callbacks,
+                                       const Common::Redis::RespValue& request);
   void updateStats(const bool success);
 
-  SplitRequestBase(CommandStats& command_stats, TimeSource& time_source)
+  SplitRequestBase(CommandStats& command_stats, TimeSource& time_source, bool latency_in_micros)
       : command_stats_(command_stats) {
-    command_latency_ms_ = std::make_unique<Stats::Timespan>(command_stats_.latency_, time_source);
+    if (latency_in_micros) {
+      command_latency_ = std::make_unique<Stats::TimespanWithUnit<std::chrono::microseconds>>(
+          command_stats_.latency_, time_source);
+    } else {
+      command_latency_ = std::make_unique<Stats::Timespan>(command_stats_.latency_, time_source);
```

Comment:
NBD, but can we use `std::make_unique<Stats::TimespanWithUnit<std::chrono::milliseconds>>` here instead of `Timespan` to make it more clear?

Reply:
yes. @ramaraochavali . It will make it very clear the time unit getting used. I will do that.